yjit/src/core.rs

```diff
@@ -411,43 +411,101 @@ impl From<Opnd> for YARVOpnd {
     }
 }
 
-/// Maximum index of stack temps that could be in a register
-pub const MAX_REG_TEMPS: u8 = 8;
+/// Number of registers that can be used for stack temps or locals
+pub const MAX_MAPPED_REGS: usize = 5;
 
-/// Bitmap of which stack temps are in a register
-#[derive(Copy, Clone, Default, Eq, Hash, PartialEq, Debug)]
-pub struct RegTemps(u8);
+/// Maximum index of stack temps or locals that could be in a register
+pub const MAX_REG_OPNDS: u8 = 8;
```

Comment:
Don't hate me but I find this constant name a bit confusing. The names gives the impression that there is a limit on the number of registers or operands that could be passed into registers somewhere.

However, this constant has the same value and is used in the same way as `MAX_TEMP_TYPES` and `MAX_LOCAL_TYPES`. So maybe we should just rename those constants `MAX_CTX_TEMPS` and `MAX_CTX_LOCALS` and use those constants instead?

Reply:
Makes sense. Renamed `MAX_TEMP_TYPES`/`MAX_LOCAL_TYPES` to `MAX_CTX_TEMPS`/`MAX_CTX_LOCALS` and used them https://github.com/ruby/ruby/pull/11157/commits/6bae029f9b8eea1b439ef03a7c08874dbac9d6de.